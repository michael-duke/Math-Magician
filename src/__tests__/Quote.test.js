const { render, screen } = require('@testing-library/react');
require('@testing-library/jest-dom');
const Quote = require('../pages/Quote').default;

describe('Tests for Quote component', () => {
  it('Render Quote page', () => {
    const quote = render(<Quote />);
    expect(quote).toMatchSnapshot();
  });

  it("Check if initial quote by 'Dean Schlicter' exists", () => {
    render(<Quote />);
    expect(
      screen.getByText(
        /Go down deep enough into anything and you will find mathematics./i,
      ),
    ).toBeInTheDocument();
  });
});
