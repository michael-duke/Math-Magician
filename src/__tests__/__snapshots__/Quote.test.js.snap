// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Quote component Render Quote page 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="quote flex flex-col items-center justify-center mt-7 p-12 space-y-10"
      >
        <figure>
          <blockquote>
            <p>
              "
              Go down deep enough into anything and you will find mathematics.
               
              "
            </p>
          </blockquote>
          <figcaption>
            —
            Dean Schlicter
          </figcaption>
        </figure>
        <button
          class="p-3 font-stange font-semibold bg-slate-600 text-white uppercase rounded-md hover:bg-white hover:text-slate-600"
          data-mdb-ripple="true"
          data-mdb-ripple-color="dark"
          type="button"
        >
          Get Random Quote
        </button>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="quote flex flex-col items-center justify-center mt-7 p-12 space-y-10"
    >
      <figure>
        <blockquote>
          <p>
            "
            Go down deep enough into anything and you will find mathematics.
             
            "
          </p>
        </blockquote>
        <figcaption>
          —
          Dean Schlicter
        </figcaption>
      </figure>
      <button
        class="p-3 font-stange font-semibold bg-slate-600 text-white uppercase rounded-md hover:bg-white hover:text-slate-600"
        data-mdb-ripple="true"
        data-mdb-ripple-color="dark"
        type="button"
      >
        Get Random Quote
      </button>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
